pkg/render/gateway_api.go

```diff
@@ -336,17 +335,31 @@ func GatewayAPIResourcesGetter() func() *gatewayAPIResources {
 
 var GatewayAPIResources = GatewayAPIResourcesGetter()
 
-func GatewayAPICRDs(log logr.Logger) []client.Object {
+func GatewayAPICRDs(provider operatorv1.Provider) ([]client.Object, []client.Object) {
 	resources := GatewayAPIResources()
-	gatewayAPICRDs := make([]client.Object, 0, len(resources.k8sCRDs)+len(resources.envoyCRDs))
+	essentialCRDs := make([]client.Object, 0, len(resources.k8sCRDs)+len(resources.envoyCRDs))
+	optionalCRDs := make([]client.Object, 0, len(resources.k8sCRDs)+len(resources.envoyCRDs))
 	for _, crd := range resources.k8sCRDs {
-		gatewayAPICRDs = append(gatewayAPICRDs, crd.DeepCopyObject().(client.Object))
+		if provider.IsOpenShift() {
+			// OpenShift 4.19+ restricts the Gateway CRDs that we can install, so report
+			// that only some of them are essential.
+			switch strings.TrimSuffix(crd.Name, ".gateway.networking.k8s.io") {
+			case "gatewayclasses", "gateways", "httproutes", "referencegrants":
```

Comment:
This is a pretty good way to check a value is in a slice of values, not sure why I never though of that before.